include/circt/Dialect/RTL/RTLStatements.td

```diff
@@ -19,9 +19,12 @@ def ConnectOp : RTLOp<"connect", [InOutTypeConstraint<"src", "dest">]> {
     ```
     }];
 
-  let arguments = (ins InOutType:$dest, RTLValueType:$src);
+  let arguments = (ins 
+    Arg<InOutType, "", [MemWrite]>:$dest,
+    Arg<RTLValueType, "", [MemRead]>:$src
```

Comment:
src is an ssa value, I don't think you need MemRead here.

Reply:
Good catch..  When I started working on this I didn't realize only values were allowed on the RHS.  Fixed!